### ISSUE TITLE
Replace gnu.io dependency of add-ons with serial transport

### ISSUE THREAD

**Issue by wborn** [openhab/openhab-addons#7573]
There are still add-ons which depend on gnu.io for their serial communications. By using the serial transport we can switch between serial implementations (rxtx, javacomm) without having to change any add-on code.

These bundles still depend on gnu.io:

* [ ] org.openhab.binding.cm11a
* [ ] org.openhab.binding.digiplex
* [ ] org.openhab.binding.dscalarm
* [ ] org.openhab.binding.elerotransmitterstick
* [ ] org.openhab.binding.jeelink
* [ ] org.openhab.binding.knx
* [ ] org.openhab.binding.lgtvserial
* [ ] org.openhab.binding.lutron
* [ ] org.openhab.binding.meteostick
* [ ] org.openhab.binding.nibeheatpump
* [ ] org.openhab.binding.oceanic
* [ ] org.openhab.binding.pentair
* [ ] org.openhab.binding.phc
* [ ] org.openhab.binding.pioneeravr
* [ ] org.openhab.binding.regoheatpump
* [ ] org.openhab.binding.rme
* [ ] org.openhab.binding.rotelra1x
* [ ] org.openhab.binding.smartmeter
* [ ] org.openhab.binding.urtsi
* [ ] org.openhab.binding.velbus
* [ ] org.openhab.io.transport.modbus

**Comment by lolodomo** [comment]
The rotelra1x binding is supposed to be removed in OH 3.0 as it was replaced by the rotel binding. We kept it in 2.5 to let the time to users to move to the new binding.

**Comment by wborn** [comment]
Great! I've updated the list so we won't spend any time on migrating that one. :+1:

**Comment by gnlpfjh** [comment]
With #6447 phc binding uses serial transport.

**Comment by wborn** [comment]
Very nice! I even see you updated the POM for the import that was added in https://github.com/openhab/openhab-addons/pull/7589. :+1:

**Comment by mvalla** [comment]
What is to be done in case of a transitive dependency from a library that uses nrjavaserial (like in the case of KNX) ?

**Comment by bodiroga** [comment]
This should be pinned again, I think that @jsjames has unpinned it unintentionally :wink:

**Comment by holgerfriedrich** [comment]
@wborn I could need some advice on how to proceed with KNX binding  (Calimero library). We could easily remove the calimero-rxtx library, but the would also remove the KNX over serial interfacte feature. So this is not an option. Rewriting the RxtxAdapter  class as suggested could be an option, but we would need to do it in the context ""tuwien.auto.calimero.serial", as a class lookup is hard coded in the calimero-core library as follows:
`src/tuwien/auto/calimero/serial/LibraryAdapter.java:                    final Class<?> c = Class.forName("tuwien.auto.calimero.serial.RxtxAdapter");
`
Not sure if this is a good idea (technically). 
Reusing the source code for RxtxAdapter might be a problem as well, as we might run into issues with incompatible licenses (Calimero is GPL2 with classpath exception, openHab EPL2).

**Comment by kaikreuzer** [comment]
@wborn Do you have any good idea wrt @holgerfriedrich's question on how to best proceed for KNX?

**Comment by holgerfriedrich** [comment]
@kaikreuzer Calimero 2.5.1 has been released just a few hours ago; the code for serial port has changed..... let's see....

**Comment by wborn** [comment]
Yes it looks like you can now create your own `SerialCom` implementation and use that instead of the `calimero-rxtx` dependency.  It will create/use a `SerialCom` instance based on the class name in `META-INF/services/tuwien.auto.calimero.serial.spi.SerialCom`, see: https://github.com/calimero-project/calimero-rxtx/commit/1e74d815b05ef1a14268095df2b512e760305d09

**Comment by lsiepel** [comment]
Work for the Pentair binding is on the way as @jsjames filed a PR for it: https://github.com/openhab/openhab-addons/pull/13485

**Comment by holgerfriedrich** [comment]
@wborn I created our own ```SerialCom``` implementation, but have a problem once the framework reloads the KNX plugin. ```ServiceLoader``` does not find the implementation (or I have not yet found a way to properly push it), so I need to restart openHAB to properly reload.
If you have any suggestion, please comment in #14051.

**Comment by mvalla** [comment]
For [openwebnet] just sbmitted a new PR that makes the binding independent from `gnu.io`.
See #16376.
